File: luaui/parsers/locale/LocaleParser.lua
--[[
    LocaleParser.
    Faithful port of haxe_ui.parsers.locale.LocaleParser.
--]]

local LocaleParser = {}
LocaleParser.__index = LocaleParser

LocaleParser._parsers = {}

function LocaleParser.new()
    local self = setmetatable({}, LocaleParser)
    return self
end

function LocaleParser:parse(data)
    error("Locale parser not implemented!")
end

function LocaleParser.get(extension)
    LocaleParser.defaultParsers()
    local cls = LocaleParser._parsers[extension]
    if cls == nil then return nil end
    return cls.new()
end

function LocaleParser.defaultParsers()
    -- register defaults here
end

function LocaleParser.register(extension, cls)
    LocaleParser._parsers[extension] = cls
end

return LocaleParser

File: flutter_ui/lib/parsers/locale/locale_parser.dart
class LocaleParser {
  static final Map<String, Type> _parsers = {};

  static LocaleParser? get(String extension) {
    Type? cls = _parsers[extension];
    if (cls == null) return null;
    // In a real framework, we'd use a registry or factory for instantiation
    return null;
  }

  static void register(String extension, Type cls) {
    _parsers[extension] = cls;
  }

  Map<String, String> parse(String data) {
    return {};
  }
}
